### DISCUSSION TITLE
Delete Pooler resource via foreground ArgoCD pruning

### DISCUSSION THREAD

**Discussion by robin5210** [cloudnative-pg/cloudnative-pg#4010] [Q&A]
With the default ArgoCD prune policy of foreground pruning, Pooler resources are stuck in an infinite loop as the CNPG controller recreates the pods before ArgoCD can delete the resource. 

Is this a bug with the Pooler as a finalizer has been set on the Pooler resource by ArgoCD?

**Comment by mjlshen** [comment]
I filed an issue for this in https://github.com/cloudnative-pg/cloudnative-pg/issues/11360 and have a proposed fix in https://github.com/cloudnative-pg/cloudnative-pg/pull/11359

**Comment by mou** [comment]
Regarding the issue, many users are affected. The issue itself is pretty severe since it affects not only CD systems but also any deletion operations that are implemented with a finalizer in place, like `kubectl delete ... --cascade=foreground`
